### PR TITLE
Remove the oeedl_file cmake function

### LINKED ISSUE #2325: Stop using oeedl_file.cmake
It's an ancient script that we're trying to deprecate, but it's a lot of rote work to remove. It's not a great user experience because (a) it's not what we instruct consumers of the SDK to use and (b) it does a bunch of "magic" that is best done explicitly in each CMake file.

See the [CMake docs](https://github.com/openenclave/openenclave/blob/master/cmake/sdk_cmake_targets_readme.md#running-oeedger8r) on what to use instead (but use `edger8r` instead of `openenclave::edger8r` as that's the name used internally).

### PR DESCRIPTION
**Pull request by jhand2**
The oeedl_file function is inflexible and is too large a function to justify how little it abstracts. Remove the function and instead use a cmake custom command to call edger8r for each enclave.

Fixes #2325

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
-include(oeedl_file)
```
Comment (radhikaj) [inline tests/CMakeLists.txt]: Can this cmake module be deleted from the repo? Or does it continue to serve some purpose?

Comment (jhand2) [reply tests/CMakeLists.txt]: Oh, will do. I intended to remove it but I guess I forgot :smile:

**Comment by jhand2** [comment]
Got canceled for some reason

bors r+

**Comment by nonpolarity** [comment]
This is a really huge project that almost all tests are related.

**Comment by jhand2** [comment]
> This is a really huge project that almost all tests are related.

Not sure what you mean

**Comment by nonpolarity** [comment]
> 
> 
> > This is a really huge project that almost all tests are related.
> 
> Not sure what you mean.
> 
> As for the motivation, this function has been considered deprecated for a while and people have been asked not to use it in new PRs. We also don't recommend that people use this externally.
> 
> The function is overly complicated, doesn't work in a number of cases, and hides what is actually happening for people who are new to the project. People generally look to test to figure out how they should write and compile their application. We shouldn't use internal only mechanisms for tests and the added code is very minimal.

Sorry I just mean a lot of work. Until oeedl_file is removed totally some developers may spread oeedl_file to new tests. Nice to see oeedl_file will become extinct.

**Comment by jhand2** [comment]
bors cancel

**Comment by jhand2** [comment]
@nonpolarity Ah I misunderstood 😄 Thanks! Not sure whats going on with bors but hopefully I can merge soon.

**Comment by BRMcLaren** [comment]
bors try

**Comment by BRMcLaren** [comment]
The B.O.R.S try was in regards to a potential issue with our CI/CD. Double checked, seemed to have been an infra timeout and nothing to be done. Please let me know if it happens again.

**Comment by jhand2** [comment]
Thanks @BRMcLaren 

bors r+

**Comment by andschwa** [comment]
Wow thanks so much @jhand2!

**Pull request merged**